c_glib/test/test-fixed-size-list-data-type.rb

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestFixedSizeListDataType < Test::Unit::TestCase
+  sub_test_case(".new") do
+    def setup
+      @value_type = Arrow::BooleanDataType.new
+      @list_size = 5
+      @field_name = "bool_field"
+    end
+
+    def test_field
+      field = Arrow::Field.new(@field_name, @value_type)
+      data_type = Arrow::FixedSizeListDataType.new(field, @list_size);
+      assert_equal(Arrow::Type::FIXED_SIZE_LIST, data_type.id);
+    end
+
+    def test_data_type
+      data_type = Arrow::FixedSizeListDataType.new(@value_type, @list_size);
+      assert_equal(Arrow::Type::FIXED_SIZE_LIST, data_type.id);
```

Comment:
We should also check `data_type.field.data_type` (or `data_type.field`).

Reply:
Thanks. Changed to the below

```ruby
      assert_equal([Arrow::Type::FIXED_SIZE_LIST, "fixed_size_list<item: bool>[5]"],
                   [data_type.id, data_type.to_s])
```